### ISSUE TITLE
Test suite failures

### ISSUE THREAD

**Issue by Apteryks** [xwmx/pandoc-ruby#39]
With the following GEM_PATH:
`/gnu/store/8ql9jjzl8q291ghsxlkm1wn5bpdvgcqw-ruby-2.6.5/lib/ruby/vendor_ruby:/gnu/store/2cg77jha3hq8s497kr6g88q5mrba70sq-ruby-mocha-1.11.2/lib/ruby/vendor_ruby`

And with ghc-pandoc 2.7.3, I get the following test suite failures:

```
starting phase `check'
/gnu/store/8ql9jjzl8q291ghsxlkm1wn5bpdvgcqw-ruby-2.6.5/bin/ruby -w -I"lib:lib:test" -I"/gnu/store/8ql9jjzl8q291ghsxlkm1wn5bpdvgcqw-ruby-2.6.5/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib" "/gnu/store/8ql9jjzl8q291ghsxlkm1wn5bpdvgcqw-ruby-2.6.5/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/rake_test_loader.rb" "test/test_conversions.rb" "test/test_pandoc_ruby.rb" 
Run options: --seed 2995

# Running:

F...................................S..................................................................................................F........................F..........F.F.......FF...F.......F......

Finished in 8.310657s, 24.1858 runs/s, 37.4218 assertions/s.

  1) Failure:
PandocRuby#test_0011_converts underscore symbol args to hyphenated long options [/tmp/guix-build-ruby-pandoc-ruby-2.1.4.drv-0/source/lib/pandoc-ruby.rb:253]:
unexpected invocation: #<PandocRuby:0xd36008>.execute("/gnu/store/qdqjmpbkk9majih7vws4avlwjz29ss1c-ghc-pandoc-2.7.3/bin/pandoc --email-obfuscation javascript --table-of-contents")
unsatisfied expectations:
- expected exactly once, invoked never: #<PandocRuby:0xd36008>.execute("pandoc --email-obfuscation javascript --table-of-contents")


  2) Failure:
Conversions#test_0073_converts latex to plain [/tmp/guix-build-ruby-pandoc-ruby-2.1.4.drv-0/source/test/test_conversions.rb:26]:
--- expected
+++ actual
@@ -1,4 +1,3 @@
-"THIS IS A TITLE
+"This is a Title
 
-
-Some _emphasized text_ and a link"
+Some emphasized text and a link"


  3) Failure:
Conversions#test_0013_converts markdown to plain [/tmp/guix-build-ruby-pandoc-ruby-2.1.4.drv-0/source/test/test_conversions.rb:26]:
--- expected
+++ actual
@@ -1,4 +1,3 @@
-"THIS IS A TITLE
+"This is a Title
 
-
-Some _emphasized text_ and a link"
+Some emphasized text and a link"


  4) Failure:
Conversions#test_0022_converts html to beamer [/tmp/guix-build-ruby-pandoc-ruby-2.1.4.drv-0/source/test/test_conversions.rb:26]:
--- expected
+++ actual
@@ -1,7 +1,5 @@
 "\\begin{frame}{This is a Title}
 \\protect\\hypertarget{this-is-a-title}{}
-
 Some \\emph{emphasized text} and
 \\href{http://daringfireball.net/projects/markdown/}{a link}
-
 \\end{frame}"


  5) Failure:
Conversions#test_0002_converts markdown to beamer [/tmp/guix-build-ruby-pandoc-ruby-2.1.4.drv-0/source/test/test_conversions.rb:26]:
--- expected
+++ actual
@@ -1,7 +1,5 @@
 "\\begin{frame}{This is a Title}
 \\protect\\hypertarget{this-is-a-title}{}
-
 Some \\emph{emphasized text} and
 \\href{http://daringfireball.net/projects/markdown/}{a link}
-
 \\end{frame}"


  6) Failure:
Conversions#test_0062_converts latex to beamer [/tmp/guix-build-ruby-pandoc-ruby-2.1.4.drv-0/source/test/test_conversions.rb:26]:
--- expected
+++ actual
@@ -1,7 +1,5 @@
 "\\begin{frame}{This is a Title}
 \\protect\\hypertarget{this-is-a-title}{}
-
 Some \\emph{emphasized text} and
 \\href{http://daringfireball.net/projects/markdown/}{a link}
-
 \\end{frame}"


  7) Failure:
Conversions#test_0033_converts html to plain [/tmp/guix-build-ruby-pandoc-ruby-2.1.4.drv-0/source/test/test_conversions.rb:26]:
--- expected
+++ actual
@@ -1,4 +1,3 @@
-"THIS IS A TITLE
+"This is a Title
 
-
-Some _emphasized text_ and a link"
+Some emphasized text and a link"


  8) Failure:
Conversions#test_0053_converts rst to plain [/tmp/guix-build-ruby-pandoc-ruby-2.1.4.drv-0/source/test/test_conversions.rb:26]:
--- expected
+++ actual
@@ -1,4 +1,3 @@
-"THIS IS A TITLE
+"This is a Title
 
-
-Some _emphasized text_ and a link"
+Some emphasized text and a link"


  9) Failure:
Conversions#test_0042_converts rst to beamer [/tmp/guix-build-ruby-pandoc-ruby-2.1.4.drv-0/source/test/test_conversions.rb:26]:
--- expected
+++ actual
@@ -1,7 +1,5 @@
 "\\begin{frame}{This is a Title}
 \\protect\\hypertarget{this-is-a-title}{}
-
 Some \\emph{emphasized text} and
 \\href{http://daringfireball.net/projects/markdown/}{a link}
-
 \\end{frame}"


201 runs, 311 assertions, 9 failures, 0 errors, 1 skips

You have skipped tests. Run with --verbose for details.
rake aborted!
Command failed with status (1): [ruby -w -I"lib:lib:test" -I"/gnu/store/8ql9jjzl8q291ghsxlkm1wn5bpdvgcqw-ruby-2.6.5/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib" "/gnu/store/8ql9jjzl8q291ghsxlkm1wn5bpdvgcqw-ruby-2.6.5/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/rake_test_loader.rb" "test/test_conversions.rb" "test/test_pandoc_ruby.rb" ]

Tasks: TOP => test
(See full trace by running task with --trace)
command "rake" "test" failed with status 1
```

**Comment by Apteryks** [comment]
Another one I get after the one posted above is:
```
  1) Failure:
PandocRuby#test_0012_uses second arg as option [/tmp/guix-build-ruby-pandoc-ruby-2.1.4.drv-0/source/lib/pandoc-ruby.rb:253]:
unexpected invocation: #<PandocRuby:0xd27a08>.execute("/gnu/store/qdqjmpbkk9majih7vws4avlwjz29ss1c-ghc-pandoc-2.7.3/bin/pandoc --toc")
unsatisfied expectations:
- expected exactly once, invoked never: #<PandocRuby:0xd27a08>.execute("pandoc --toc")


193 runs, 303 assertions, 1 failures, 0 errors, 1 skips

You have skipped tests. Run with --verbose for details.
rake aborted!
Command failed with status (1): [ruby -w -I"lib:lib:test" -I"/gnu/store/8ql9jjzl8q291ghsxlkm1wn5bpdvgcqw-ruby-2.6.5/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib" "/gnu/store/8ql9jjzl8q291ghsxlkm1wn5bpdvgcqw-ruby-2.6.5/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/rake_test_loader.rb" "test/test_conversions.rb" "test/test_pandoc_ruby.rb" ]

```

**Comment by Apteryks** [comment]
The second error can be safely ignored, is was due to using a custom pandoc_path as seen in the unexpected invocation.

**Comment by xwmx** [comment]
Hi @Apteryks!

The first one appears to just be the custom pandoc path tripping up the test, which expects just the executable name.

The conversion-related test failures appear to be due to the version of pandoc the tests are running with, version 2.7.3. The conversion test files work with pandoc version 2.9.2+ (though might need updates for the recently-released 2.10). You can see in the last change for the test files, [c0b68df](https://github.com/xwmx/pandoc-ruby/commit/c0b68df43b2bd15568fe52b68cfe0fedd0979299), that the changes were the same strings causing these test failures.

I recommend updating pandoc to the latest version. However, since these are the only failures, it looks like pandoc is working properly and these failures are just artifacts of the testing configuration, so pandoc-ruby should work fine your configuration even with the test failures.

Let me know if you run into any other issues or have any other questions related to this!

**Comment by Apteryks** [comment]
Hi @xwmx, thanks for the prompt response! I've added a note in the package definition so that the current disabling of some tests can be cleaned up after the Guix ghc-pandoc package gets upgraded to 2.9.2+.

Thanks!
